scripts/create-darwin-volume.sh

```diff
@@ -83,6 +61,64 @@ suggest_report_error(){
     echo "       please report this @ https://github.com/nixos/nix/issues" >&2
 }
 
+generate_mount_command(){
+    if test_filevault_in_use; then
+        printf "    <string>%s</string>\n" /bin/sh -c '/usr/bin/security find-generic-password -a "Nix Volume" -w | /usr/sbin/diskutil apfs unlockVolume "Nix Volume" -mountpoint /nix -stdinpassphrase'
+    else
+        printf "    <string>%s</string>\n" /usr/sbin/diskutil mount -mountPoint /nix "Nix Volume"
+    fi
+}
+
+generate_mount_daemon(){
+    cat << EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+  <key>RunAtLoad</key>
+  <true/>
+  <key>KeepAlive</key>
+  <dict>
+    <key>PathState</key>
+    <dict>
+      <key>/nix/var/nix</key>
+      <false/>
+    </dict>
+  </dict>
+  <key>Label</key>
+  <string>org.nixos.darwin-store</string>
+  <key>ProgramArguments</key>
+  <array>
+$(generate_mount_command)
+  </array>
+</dict>
+</plist>
+EOF
+}
+
+prepare_darwin_volume_password(){
+    sudo /usr/bin/expect -f - "$1" "$2" << 'EOF'
+log_user 0
+set VOLUME [lindex $argv 0];
+set UUID [lindex $argv 1];
+set PASSPHRASE [exec /usr/bin/ruby -rsecurerandom -e "puts SecureRandom.hex(32)"]
```

Comment:
`/usr/bin/ruby` is deprecated on macOS (all scripting languages besides bash are; python3 is available if and only if Xcode is installed).

I think bash version of this hex generation could look like

```sh
cat /dev/random | head -c 32 | xxd -p -c 256
```